openassessment/xblock/openassessmentblock.py

```diff
@@ -200,22 +201,45 @@ class OpenAssessmentBlock(
         help="Indicates whether or not there are peers to grade."
     )
 
-    def get_student_item_dict(self):
+    @property
+    def course_id(self):
+        return self._serialize_opaque_key(self.xmodule_runtime.course_id)  # pylint:disable=E1101
+
+    def get_anonymous_user_id(self, username, course_id):
+        """
+        Get the anonymous user id from Xblock user service.
+
+        Args:
+            username(str): user's name entered by staff to get info.
+            course_id(str): course id.
+
+        Returns:
+            A unique id for (user, course) pair
+        """
+        return self.runtime.service(self, 'user').get_anonymous_user_id(username, course_id)
+
+    def get_student_item_dict(self, anonymous_user_id=None):
         """Create a student_item_dict from our surrounding context.
 
         See also: submissions.api for details.
 
+        Args:
+            anonymous_user_id(str):
         Returns:
             (dict): The student item associated with this XBlock instance. This
                 includes the student id, item id, and course id.
         """
 
         item_id = self._serialize_opaque_key(self.scope_ids.usage_id)
+
         # This is not the real way course_ids should work, but this is a
         # temporary expediency for LMS integration
         if hasattr(self, "xmodule_runtime"):
-            course_id = self._serialize_opaque_key(self.xmodule_runtime.course_id)  # pylint:disable=E1101
-            student_id = self.xmodule_runtime.anonymous_student_id  # pylint:disable=E1101
+            course_id = self.course_id  # pylint:disable=E1101
+            if anonymous_user_id:
```

Comment:
I'm a little worried about what it means to pass in an anonymous user id versus pulling it from the modulestore. What would this fallback look like in production?